File: app/src/main/java/com/koala/infinitum/android_project/mainFragments/list/ListAdapter.java
package com.koala.infinitum.android_project.mainFragments.list;

import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.koala.infinitum.android_project.R;

import java.util.ArrayList;

class ListAdapter extends RecyclerView.Adapter<ListItemHolder> {
    private ArrayList<String> items;

    ListAdapter(ArrayList<String> items) {
        super();
        this.items = items;
    }

    @Override
    public ListItemHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item, parent, false);
        return new ListItemHolder(view);
    }

    @Override
    public void onBindViewHolder(ListItemHolder holder, int position) {
        holder.update(items.get(position));
        holder.setOnItemClickListener(new OnItemClickListener() {
            @Override
            public void onItemClick(View v, int pos) {
                Log.d("custom-logs", pos + "");
            }
        });
    }

    @Override
    public int getItemCount() {
        return this.items.size();
    }
}

File: app/src/main/java/com/koala/infinitum/android_project/mainFragments/list/ListAdapter.kt
package com.koala.infinitum.android_project.mainFragments.list

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.LinearLayout
import com.koala.infinitum.android_project.R

//Адаптер для списка, отслеживает измененые viewHolder и меняет элементы в их layout
class ListAdapter(private val items: ArrayList<String>) : RecyclerView.Adapter<ListItemHolder>() {
    override fun getItemCount(): Int = items.toArray().count()

    override fun onBindViewHolder(holder: ListItemHolder?, position: Int) {
        holder?.update(items[position])
    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ListItemHolder {
        val itemHolder = LayoutInflater.from(parent!!.context).inflate(R.layout.list_item, parent, false) as LinearLayout
        return ListItemHolder(itemHolder)
    }
}
